tests/load/locustfile.py

```diff
@@ -4,42 +4,98 @@
+    locust CLI.
+    """
+    if isinstance(value, bool):
+        return value
+    if value.lower() in {"true", "1", "yes"}:
+        return True
+    elif value.lower() in {"false", "0", "no"}:
+        return False
+    else:
+        raise ValueError(f"Invalid boolean value: {value}")
+
+
+@locust.events.init_command_line_parser.add_listener
+def _(parser):
+    parser.add_argument(
+        "--records-to-link", type=int, default=100, help="Number of records to link in the test run"
+    )
+    parser.add_argument(
+        "--link-probability",
+        type=float,
+        default=0.5,
+        help="Probability of linking a record",
+    )
+
+    parser.add_argument(
+        "--seeded",
+        type=str_to_bool,
+        default=False,
+        help="Indicates if the database has been seeded with initial data",
+    )
+
+
 class LoadTest(locust.HttpUser):
     seed_data = None
-    seeded = False
     seed_lock = threading.Lock()
 
-    def load_seed_data(self):
-        locust_file = pathlib.Path(__file__).resolve()
-        project_root = locust_file.parents[2]
-        json_path = os.path.join(project_root, "tests/load/assets/test_data.json")
-
-        with open(json_path, "r") as f:
-            return json.load(f)
-
     def on_start(self):
+        # Check if the database has been seeded before starting the test
+        self.__class__.seeded = self.environment.parsed_options.seeded
+
         with self.seed_lock:
             if not self.__class__.seeded:
-                self.__class__.seed_data = self.load_seed_data()
-                # TODO: Adjust to loop through the seed data; API endpoint has a limit of 100 clusters
-                self.client.post("/api/seed", json=self.__class__.seed_data)
-                self.__class__.seeded = True
+                locust_file = pathlib.Path(__file__).resolve()
+                project_root = locust_file.parents[2]
+                seed_data_path = os.path.join(project_root, "tests/load/assets/seed_data.json")
+                with open(seed_data_path, "rb") as input_file:
+                    clusters_iter = ijson.items(input_file, "clusters.item", use_float=True)
+                    chunk = []
+                    chunk_size = 100
+                    for cluster in clusters_iter:
+                        chunk.append(cluster)
+                        if len(chunk) == chunk_size:
+                            seed_data = json.dumps({"clusters": chunk}, indent=2)
+                            self.client.post("/api/seed", seed_data)
+                            chunk = []
 
-    def pick_random_record(self):
-        # Pick a random record from the seed data
-        rando = random.choice(self.__class__.seed_data["clusters"])
-        return rando
+                    if chunk:
+                        seed_data = json.dumps({"clusters": chunk}, indent=2)
+                        self.client.post("/api/seed", seed_data)
+                self.__class__.seeded = True
```

Comment:
I'm new to locust, but does this value need to be persisted back to `self.environment`?  I'm not seeing how `self.environment.parsed_options.seeded` would return True on subsequent runs.

Reply:
wfm